libavcodec/qsvdec.c

```diff
@@ -514,8 +512,6 @@ static int qsv_decode(AVCodecContext *avctx, QSVContext *q,
         bs.DataLength = avpkt->size;
         bs.MaxLength  = bs.DataLength;
         bs.TimeStamp  = PTS_TO_MFX_PTS(avpkt->pts, avctx->pkt_timebase);
-        if (avctx->field_order == AV_FIELD_PROGRESSIVE)
-            bs.DataFlag   |= MFX_BITSTREAM_COMPLETE_FRAME;
     }
```

Comment:
Personally, I think it's better to do addition instead of subtraction. Any possible to modify avctx.field_order in parser?

Reply:
This flag is not only about field_order. We need to make sure the packet is complete as well. FFmpeg may split a packet into several packets and ffmpeg need to parse next pkt to know if it is complete. 
I submit related issue to MSDK: https://github.com/Intel-Media-SDK/MediaSDK/issues/2618.

Reply:
If you know any flag indicating whether this pkt contain whole frame, it will be helpful to me. : D

Reply:
OK, I got your point. PROGRESSIVE doen't mean COMPLETE. There should not have this kind of flag to indicate complete frame. For av1, COMPLETE is guaranteed by av1_parser. Like my patch:
https://patchwork.ffmpeg.org/project/ffmpeg/patch/20211203080920.1948453-1-fei.w.wang@intel.com/
You can take a look if this suit for your codec/parser.